src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionsConfigExtensions.cs

```diff
@@ -48,6 +48,9 @@ public static class MessageDrivenSubscriptionsConfigExtensions
         {
             Guard.AgainstNullAndEmpty(nameof(publisherEndpoint), publisherEndpoint);
 
+            if (publisherEndpoint.Contains("@"))
+                throw new ArgumentException($"Expected an endpoint name but received '{publisherEndpoint}'.");
```

Comment:
should that be something like "Character '@' is invalid for a logical endpoint name. In case you want to specify an address instead of an endpoint name, consider using the instance mapping file for MSMQ or refer to the routing documentation"?